src/frontend/src/scheduler/distributed/query_manager.rs

```diff
@@ -132,6 +144,32 @@ impl QueryResultFetcher {
             yield DataChunk::from_protobuf(response?.get_record_batch()?)?;
         }
     }
+
+    fn run(self) -> BoxedDataChunkStream {
+        Box::pin(self.run_inner())
+    }
+
+    // #[try_stream(ok = DataChunk, error = RwError)]
+    async fn run_wrapper(self, shutdown_rx: Receiver<u8>) -> impl DataChunkStream {
```

Comment:
Tried to build a wrapper on original function and use async-stream's stream! macro.